platform/storage/zephyr/nvs/src/mender-storage.c

```diff
@@ -346,7 +346,7 @@ mender_storage_save_update_state(mender_update_state_t state, const char *artifa
 
     size_t artifact_type_size;
 
-    if (!checked_nvs_write(&mender_storage_nvs_handle, MENDER_STORAGE_NVS_UPDATE_STATE, (const void *)state, sizeof(state))) {
+    if (!checked_nvs_write(&mender_storage_nvs_handle, MENDER_STORAGE_NVS_UPDATE_STATE, (const void *)&state, sizeof(state))) {
```

Comment:
I think you don't need the `void*` cast anymore, the missing `&` was the real issue here.

Reply:
Thanks! That makes sense